R/pkg/R/utils.R

```diff
@@ -339,20 +339,40 @@ varargsToEnv <- function(...) {
 varargsToStrEnv <- function(...) {
   pairs <- list(...)
   env <- new.env()
-  for (name in names(pairs)) {
-    value <- pairs[[name]]
-    if (!(is.logical(value) || is.numeric(value) || is.character(value) || is.null(value))) {
-      stop(paste0("Unsupported type for ", name, " : ", class(value),
-           ". Supported types are logical, numeric, character and NULL."))
-    }
-    if (is.logical(value)) {
-      env[[name]] <- tolower(as.character(value))
-    } else if (is.null(value)) {
-      env[[name]] <- value
-    } else {
-      env[[name]] <- as.character(value)
+  nameList <- names(pairs)
+  ignoredNames <- list()
+  i <- 1
```

Comment:
Oh, sure. Actually, it took me a while to find out a clean way but it ended up with introducing another variable here...

```r
value <- pairs[[name]]
```

is failed when `name` is empty string (when they are non-named arguments), producing the error as below:

```r
Error in env[[name]] <- value :
  zero-length variable name
```

I wanted to access to that problematic values to print in 

```r
ignoredNames <- append(ignoredNames, pairs[i])
...
warning(paste0("Non-named arguments ignored: ", paste(ignoredNames, collapse = ", "), "."),
         call. = FALSE)
```

To cut it short, I introduced that variable to access to the value when `name` is empty. Maybe there should be another cleaver way but I couldn't come up with it.